File: practice-algos/left_rotation/gpt_sol.js
function leftRotate(arr, d) {
    let n = arr.length;
    d %= n;  // This line handles cases where d > n
    return arr.slice(d).concat(arr.slice(0, d));
}

// Example usage
let arr = [1, 2, 3, 4, 5];
let d = 2;
let result = leftRotate(arr, d);
console.log("Rotated Array:", result);

File: practice-algos/left_rotation/gpt_sol.py
def left_rotate(arr, d):
    n = len(arr)
    d = d % n  # This line handles cases where d > n
    return arr[d:] + arr[:d]

# Example usage
arr = [1, 2, 3, 4, 5]
d = 2
result = left_rotate(arr, d)
print("Rotated Array:", result)
